This is a modified version of test_processes

loumalouomega opened the pull request:
This version is correcting a minor error that appeared after the modification of the model_part_io, when the model_part_io.mda was change to model_part_io_read.mdpa

roigcarlo commented on kratos/tests/geometries/test_triangle_2d3.cpp:
```diff
@@ -0,0 +1,545 @@
+     * @param  ThisMethod Input Integration method
+     * @return            String with the name of the input integration method
+     */
+    std::string GetIntegrationName(Geometry< Node<3> >& geom, Geometry<Node<3> >::IntegrationMethod ThisMethod) {
```
Geometry does not have any use here

pooyan-dadvand replied:
Actually I don't have any clue why it is passed to it! @RiccardoRossi  do you have any idea?

roigcarlo commented on kratos/tests/geometries/test_triangle_2d3.cpp:
```diff
@@ -0,0 +1,545 @@
+        error_msg << "Geometry Type = " << GetGeometryName(geom) << " - IntegrationMethod = " << GetIntegrationName(geom,ThisMethod) << " -- the integration method is not supported " << std::endl;
+      } else {
+        unsigned int strain_size;
+
```
Couldn't this be:

```c++
strain_size = 3 * dim;
```

roigcarlo commented on kratos/tests/geometries/test_triangle_2d3.cpp:
```diff
@@ -0,0 +1,545 @@
+      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_3, ExpectedArea, error_msg));
+      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_4, ExpectedArea, error_msg));
+      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_5, ExpectedArea, error_msg));
+
```
I am not sure what is being tested here since ```VerifyStrainExactness``` returns nothing

roigcarlo commented on kratos/tests/geometries/test_triangle_2d3.cpp:
```diff
@@ -0,0 +1,545 @@
+      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_2, ExpectedArea, error_msg));
+      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_3, ExpectedArea, error_msg));
+      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_4, ExpectedArea, error_msg));
+      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_5, ExpectedArea, error_msg));
```
Also, should we divide this into different tests?

What happens if `GI_GAUSS_1` fails but the rest work?

jcotela replied:
It would be good to have a discussion on this, since mostly everything in geometry depends on integration method. Shoud we do individual tests?

RiccardoRossi commented:
well, for this there is also a python test, so please take care not to
break that if you change somethign

regarding strain:

1D case - strain has size 1
2D case - strain has size 3
3D case - strain has size 6,

the formula 3*Dim does not work ...

concerning the line

line2D3, is a line with 3 nodes...

cheers
Riccardo

On Tue, Jan 17, 2017 at 11:13 AM, Carlos Roig <notifications@github.com>
wrote:

> *@roigcarlo* commented on this pull request.
> ------------------------------
>
> In kratos/tests/geometries/test_triangle_2d3.cpp
> <https://github.com/KratosMultiphysics/Kratos/pull/6#pullrequestreview-16958015>
> :
>
> > +    KRATOS_TEST_CASE_IN_SUITE(TestTriangle2D3GaussPoints, KratosCoreGeometriesFastSuite) {
> +      auto geom = GenerateTriangle2D3<Node<3>>();
> +
> +      std::stringstream error_msg;
> +
> +      boost::numeric::ublas::bounded_matrix<double,4,3> DN_DX;
> +      array_1d<double,4> N;
> +      double ExpectedArea;
> +
> +      GeometryUtils::CalculateGeometryData(*geom, DN_DX, N, ExpectedArea);
> +
> +      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_1, ExpectedArea, error_msg));
> +      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_2, ExpectedArea, error_msg));
> +      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_3, ExpectedArea, error_msg));
> +      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_4, ExpectedArea, error_msg));
> +      KRATOS_CHECK(VerifyAreaByIntegration(*geom, GeometryData::GI_GAUSS_5, ExpectedArea, error_msg));
>
> Also, should we divide this into different tests?
>
> What happens if GI_GAUSS_1 fails but the rest work?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/KratosMultiphysics/Kratos/pull/6#pullrequestreview-16958015>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHr7EfceSRleInZAWGR_SjCQnOaqKTPJks5rTJRjgaJpZM4LlbiD>
> .
>



-- 


*Riccardo Rossi*

PhD, Civil Engineer


member of the Kratos Team: www.cimne.com/kratos

Tenure Track Lecturer at Universitat Politècnica de Catalunya,
BarcelonaTech (UPC)

Full Research Professor at International Center for Numerical Methods in
Engineering (CIMNE)


C/ Gran Capità, s/n, Campus Nord UPC, Ed. B0, Despatx 102

(please deliver post to the porters of building C1)

08034 – Barcelona – Spain – www.cimne.com  -

T.(+34) 93 401 56 96 skype: *rougered4*



<http://www.cimne.com/>

<https://www.facebook.com/cimne> <http://blog.cimne.com/>
<http://vimeo.com/cimne> <http://www.youtube.com/user/CIMNEvideos>
<http://www.linkedin.com/company/cimne> <https://twitter.com/cimne>

Les dades personals contingudes en aquest missatge són tractades amb la
finalitat de mantenir el contacte professional entre CIMNE i voste. Podra
exercir els drets d'accés, rectificació, cancel·lació i oposició,
dirigint-se a cimne@cimne.upc.edu. La utilització de la seva adreça de
correu electronic per part de CIMNE queda subjecte a les disposicions de la
Llei 34/2002, de Serveis de la Societat de la Informació i el Comerç
Electronic.

 Imprimiu aquest missatge, només si és estrictament necessari.
<http://www.cimne.com/>

loumalouomega committed:
This is a modified version of test_processes

This version is correcting a minor error that appeared after the modification of the model_part_io, when the model_part_io.mda was change to model_part_io_read.mdpa

pooyan-dadvand commented:
Have you verified if the tests are running?

loumalouomega commented:
Of course, at least this one, I just show that the test_KratosCore.py is failing, one sec and I patch it

pooyan-dadvand reviewed (approved):
👍

loumalouomega committed:
Adding two more files to the pull request
